### PR TITLE
Fix #20400, Start-Process with -NoNewWindow returns null ExitCode on Process object with -PassThru

### LINKED ISSUE #20400: Start-Process does not populate the process object's exit code with NoNewWindow switch
### Prerequisites

- [X] Write a descriptive title.
- [x] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I noticed that the `Start-Process` cmdlet sometimes does not populate the `ExitCode` property in the output, I believe it should.
This happed at least when `-PassThru`, `-Wait`, and `-NoNewWindow` switches are provided. `-NoNewWindow` seems to be the reason, since the exit code is available without it.

## Steps to reproduce
You can see this behavior with this simple test script
```
Set-Content -Value "exit 1" -Path script.ps1
$p = start-process "pwsh" script.ps1 -PassThru -NoNewWindow -Wait
$p.ExitCode  # Should output: 1
$p = start-process "pwsh" script.ps1 -PassThru -Wait
$p.ExitCode # Outputs: 1
```

### Expected behavior

```console
PS C:\> set-content -Value "exit 1" -Path script.ps1
PS C:\> $p = start-process "pwsh" script.ps1 -PassThru -NoNewWindow -Wait
PS C:\> $p.ExitCode
1
PS C:\> $p = start-process "pwsh" script.ps1 -PassThru -Wait
PS C:\> $p.ExitCode
1
```

### Actual behavior

```console
PS C:\> set-content -Value "exit 1" -Path script.ps1
PS C:\> $p = start-process "pwsh" script.ps1 -PassThru -NoNewWindow -Wait
PS C:\> $p.ExitCode
PS C:\> $p = start-process "pwsh" script.ps1 -PassThru -Wait
PS C:\> $p.ExitCode
1
```

### Error details

_No response_

### Environment data

```powershell
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.0-preview.5
PSEdition                      Core
GitCommitId                    7.4.0-preview.5
OS                             Microsoft Windows 10.0.20348
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by CodeCyclone**
# PR Summary

Fixing issue #20400 along with some duplicates.

This issue was introduced by change #19096.
This change moved from .NET creating process objects to use P/Invoke (Windows APIs). The code would fetch the Process object from .NET but by design it throws an exception silently and ExitCode is null because it didn't handle spawning the process itself.

Calling the Handle property on the fetched Process will resolve ExitCode state because it fetches the underlying SafeHandle used to update the ExitCode property.

## PR Context

When you call Start-Process with -NoNewWindow, ExitCode will be null. This breaks a lot of customers of PowerShell both internally and externally. 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by iSazonov** [comment]
@CodeCyclone Please add test(s) to avoid regressions.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs [hunk 1]
```diff
@@ -2126,6 +2126,9 @@ protected override void BeginProcessing()
                     jobAssigned = jobObject.AssignProcessToJobObject(processInfo.Process);
                 }
 
+                // Fetch process handle to trigger Process object to update its state by calling SetProcessHandle, discard result.
```
Comment (jborean93) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:2129]: Would be good to link to the issue so people looking at the code can see why this process handle is retrieved.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
Please add a test in https://github.com/PowerShell/PowerShell/blob/811efa46df822bf7be6179b6219f9f9d160eb7d5/test/powershell/Modules/Microsoft.PowerShell.Management/Start-Process.Tests.ps1

**Comment by CodeCyclone** [comment]
Added test, addressed feedback on commenting change. Added more detail and referenced issue.
Screenshot of the test failing without my fix:
image

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs [hunk 2]
```diff
@@ -2126,6 +2126,12 @@ protected override void BeginProcessing()
                     jobAssigned = jobObject.AssignProcessToJobObject(processInfo.Process);
                 }
 
+                // Addresses issue #20400: Start-Process does not populate the process object's exit code with NoNewWindow switch
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs:2129]: We never add links to issues.

**Review by daxian-dbw — APPROVED** [review]

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Thanks a lot

**Pull request merged**